cvmfs/catalog_mgr_rw.cc

```diff
@@ -1229,6 +1230,38 @@ void WritableCatalogManager::ScheduleCatalogProcessing(
   spooler_->ProcessCatalog(catalog->database_path());
 }
 
+/**
+ * Copy catalog to local cache.server
+ * Must be an atomic write into the local_cache_dir
+ * As such: create a temporary copy in local_cache_dir/txn and then do a 
+ * `rename` (which is atomic) to the actual cache path
+ * 
+ * @returns true on success, otherwise false
+ */
+bool WritableCatalogManager::CopyCatalogToLocalCache(
+                                          const upload::SpoolerResult &result) {
+  std::string tmp_catalog_path;
+  const std::string cache_catalog_path = local_cache_dir_ + "/"
+                                  + result.content_hash.MakePathWithoutSuffix();
+  FILE *fcatalog = CreateTempFile(local_cache_dir_ + "/txn/catalog", 0666,
+                                                      "w", &tmp_catalog_path);
+  if (!fcatalog) {
+    LogCvmfs(kLogCatalog, kLogDebug | kLogStderr,
+                               "Creating file for temporary catalog failed: %s",
+                               tmp_catalog_path.c_str());
+    return false;
+  }
+  CopyPath2File(result.local_path.c_str(), fcatalog);
```

Comment:
is this safe?
`const upload::SpoolerResult &result` is not used/reused somewhere else afterwards?
for sure some `ifs` will be added to prevent `unlink(result.local_path)` in other code sections...

Reply:
--> nevermind